src/containers/TemplateBuilder/Templates.tsx

```diff
@@ -128,21 +190,118 @@ const ExportButton: React.FC<CellProps> = ({ template: { code, version, id } })
 
   return (
     <div key="duplicate">
       <div
         className="clickable"
         onClick={async (e) => {
           e.stopPropagation()
-          if (await duplicateTemplate({ id, snapshotName })) refetch()
+          setOpen(true)
         }}
       >
         <Icon className="clickable" key="export" name="copy" />
       </div>
+      <Confirm
+        open={open}
+        // Prevent click in Input from closing modal
+        onClick={(e: any) => e.stopPropagation()}
+        content={
+          <div style={{ padding: 10, gap: 10 }} className="flex-column">
+            <p>
+              Do you want to create a new template version or a whole new template type based on
+              this template?
+              <br />
+              (New template will start with an empty version history)
+            </p>
+            <Dropdown
+              selection
+              options={[
+                { key: 'version', value: 'version', text: 'Version' },
+                { key: 'template', value: 'template', text: 'Template' },
+              ]}
+              value={selectedType}
+              onChange={(_, { value }) => setSelectedType(value as 'version' | 'template')}
+            />
+            {selectedType === 'template' && (
+              <div className="flex-row-start-center" style={{ gap: 10 }}>
+                <label>New template code:</label>
+                <Input
+                  value={newCode}
+                  onChange={(e) => {
+                    setCodeError(false)
+                    setNewCode(e.target.value)
+                  }}
+                  error={codeError}
+                />
+              </div>
+            )}
+            {isTemplateUnlocked(template) && (
+              <Checkbox
+                label="Commit current version?"
+                checked={commitCurrent}
+                onChange={(_) => setCommitCurrent(!commitCurrent)}
+              />
+            )}
+            {commitCurrent && (
+              <div className="flex-row-start-center" style={{ gap: 10 }}>
+                <label>Commit message:</label>
+                <Input
+                  value={commitMessage}
+                  onChange={(e) => setCommitMessage(e.target.value)}
+                  style={{ width: '80%' }}
+                />
+              </div>
+            )}
+          </div>
+        }
+        onCancel={() => setOpen(false)}
+        onConfirm={async () => {
+          if (selectedType === 'template' && newCode === '') {
+            setCodeError(true)
+            return
+          }
+          if (commitCurrent)
+            await updateTemplate(template as any, {
+              versionId: getTemplateVersionId(),
+              versionComment: commitMessage,
+              versionTimestamp: DateTime.now().toISO(),
+            })
+          setOpen(false)
+          const templateOptions: TemplateOptions = {
+            resetVersion: commitCurrent || !isTemplateUnlocked(template),
```

Comment:
This tells the back-end to load the duplicated template with an "in-progress" versionId (i.e. a new version) rather than copying the state of the duplicate source like it normally would by default.